tools/check-typo

```diff
@@ -127,9 +127,12 @@ IGNORE_DIRS="
     if $is_svn || $is_cmd_line; then :; else continue; fi
     svnrules=''
     if $is_svn; then
-      case `git check-attr binary "$f"` in
-        *'binary: set') continue;;
-      esac
+      # 4b825d is automatically in every repository (see cache.h)
+      # diff-tree is plumbing, so the output is stable and -\t- is the
+      # guaranteed output for a binary file only
+      EMPTY=4b825dc642cb6eb9a060e54bf8d69288fbee4904
+      git diff-tree --numstat $EMPTY HEAD -- "$f" | grep -q "^-[[:blank:]]-" \
+        && continue
```

Comment:
To me this test is arcane (maybe a link to a webpage explaining what it does would be helpful?) -- I can understand what it does, but not why you would need something so complicated. What is wrong with the previous implementation, for which the meaning is much easier to guess by reading the code?